### PR TITLE
tweak(webConfigServer): RN-1169: remove unused endpoints

### PR DESCRIPTION
**Pull request by jaskfla**
### Changes

> [!WARNING]
> I’m trusting our test coverage a wee bit here. For each endpoint removed, I’ve done a global search (particularly tupaia-web, datatrak-web, meditrak-app, lesmis) for the API path. But since some of the newer orchestration servers have endpoints with the same name, there’s a slim chance I’ve removed something that is still used **and** has no unit test.

Removes these endpoints from `web-config-server`:

- `GET` countryAccessList
- `GET` downloadFiles
- `GET` organisationUnit
- `GET` organisationUnitSearch
- `GET` logout
- `POST` login
- `POST` changePassword
- `POST` pdf

### CODE REVIEW

File: packages/web-config-server/src/apiV1/index.js [hunk 1]
```diff
@@ -39,30 +30,21 @@ const handleWith = Handler =>
...
   const api = Router();
   // mount the routes
-  api.get('/getUser', catchAsyncErrors(getUser()));
-  api.post('/login', catchAsyncErrors(login));
-  api.post('/login/oneTimeLogin', catchAsyncErrors(oneTimeLogin));
-  api.get('/logout', catchAsyncErrors(logout));
-  api.post('/signup', catchAsyncErrors(appSignup()));
-  api.post('/changePassword', catchAsyncErrors(appChangePassword()));
-  api.post('/resetPassword', catchAsyncErrors(appRequestResetPassword()));
-  api.get('/downloadFiles', catchAsyncErrors(appDownloadFiles()));
```
Comment (jaskfla) [inline packages/web-config-server/src/apiV1/index.js:49]: Looks like this one was only added in June last year, but as far as I can tell the only usage of it is in `tupaia-web`, where `tupaia-web-server` handles it (by forwarding it to `central-server`)

Comment (alexd-bes) [reply packages/web-config-server/src/apiV1/index.js:49]: I think `tupaia-web-server` might be forwarding quite a few things to `web-config-server` by using the `*` route, which I have a suspicion would include these auth routes (although looks like not the `downloadFiles` route). Would be worth explicitly forwarding anything that needs to be forwarded, and using central-server where possible if there is already logic in there for a route

**Review by alexd-bes — CHANGES_REQUESTED** [review]
I have a question around tupaia-web-server forwarding requests on, and would love if we could look into explicitly forwarding these

File: packages/tupaia-web-server/src/app/createApp.ts [hunk 2]
```diff
@@ -94,6 +94,11 @@ export async function createApp(db: TupaiaDatabase = new TupaiaDatabase()) {
     )
     .use('downloadFiles', forwardRequest(CENTRAL_API_URL, { authHandlerProvider }))
     // Forward everything else to webConfigApi
+    .use('landingPage/:landingPageUrl', forwardRequest(WEB_CONFIG_API_URL, { authHandlerProvider }))
+    .use('login/oneTimeLogin', forwardRequest(WEB_CONFIG_API_URL, { authHandlerProvider }))
+    .use('resendEmail', forwardRequest(WEB_CONFIG_API_URL, { authHandlerProvider }))
+    .use('resetPassword', forwardRequest(WEB_CONFIG_API_URL, { authHandlerProvider }))
+    .use('signup', forwardRequest(WEB_CONFIG_API_URL, { authHandlerProvider }))
     .use('*', forwardRequest(WEB_CONFIG_API_URL, { authHandlerProvider }));
```
Comment (jaskfla) [inline packages/tupaia-web-server/src/app/createApp.ts:102]: Not sure if this still belongs?

File: packages/web-config-server/src/apiV1/index.js (hunk 1 above)
Comment (jaskfla) [reply packages/web-config-server/src/apiV1/index.js:49]: Done for `tupaia-web-server`. Looks like `datatrak-web-server` doesn’t forward anything to `web-config-server` like this. There are just a couple instances where it defines its own handler that uses `ctx.services.webConfig`

File: packages/tupaia-web-server/src/app/createApp.ts (hunk 2 above)
Comment (alexd-bes) [reply packages/tupaia-web-server/src/app/createApp.ts:102]: I think it shouldn't be a catchall if we are sure that all the endpoints used are covered

File: packages/web-config-server/src/apiV1/index.js [hunk 3]
```diff
@@ -26,6 +32,7 @@ export const getRoutesForApiV1 = () => {
   api.get('/getUser', catchAsyncErrors(getUser()));
   api.post('/login/oneTimeLogin', catchAsyncErrors(oneTimeLogin));
   api.post('/signup', catchAsyncErrors(appSignup()));
+  api.post('/resetPassword', catchAsyncErrors(appRequestResetPassword()));
```
Comment (alexd-bes) [inline packages/web-config-server/src/apiV1/index.js:35]: So weird that this route even worked before?!

**Review by alexd-bes — APPROVED** [review]
Looks good, thanks for tidying those up! I think we can remove the catchall in Tupaia Web Server if we have all the correct routes in there as it stands - maybe doing some reasonably thorough dev testing before merging would be good?

File: packages/web-config-server/src/apiV1/index.js (hunk 3 above)
Comment (jaskfla) [reply packages/web-config-server/src/apiV1/index.js:35]: To be honest, I hadn’t tested it. At the time I was just trusting the unit tests would catch regressions

I think I’ll need to do some serious manual testing before merging this one

**Pull request merged**